src/logger.py

```
@@ -27,30 +26,32 @@ def __init__(self, filename: str = "logs/llm_requests.json"):
-                        headers[key] = "REDACTED"
-                log_entry["extra_headers"] = headers
+                        headers_copy[key] = "REDACTED"
 
-            log_message = json.dumps(log_entry, default=str, ensure_ascii=False, indent=4)
+                log_entry["extra_headers"] = headers_copy
 
+            log_message = json.dumps(log_entry, default=str, ensure_ascii=False, indent=4)
             self.logger.info(log_message)
         except Exception as e:
-            logging.error(f"Failed to log LLM request: {e}")
+            logging.exception(f"Failed to log LLM request: {e}")
```

Ruff (0.14.10)

`exception()` call on root logger

(LOG015)